### DISCUSSION TITLE
Implement in angular

### DISCUSSION THREAD

**Discussion by techInnovator1999** [SVG-Edit/svgedit#914] [Q&A]
### Discussed in https://github.com/SVG-Edit/svgedit/discussions/906

<div type='discussions-op-text'>

<sup>Originally posted by **HussnainBSCS** July  3, 2023</sup>
HI! I am working on an angular project and I need to implement this svgedit in my angular project, plz elaborate steps how can I integrate it in angular.</div>

**Comment by AnirudhGudugunti** [comment]
Hello hussnain,i am also working on angular project and have same issue. So, did u find any method/procedure that how to integrate svg-editor in angular ,if  yes plz share here. Thank You!

**Reply by techInnovator1999** [reply]
@AnirudhGudugunti , not a proper solution yet

**Reply by KunioHaha** [reply]
Im trying add this  .d.ts, but does not work.
declare module '@svgedit/svgcanvas' {
    const svgCanvas: any;
    export default svgCanvas;
}

**Comment by w8w8w8** [comment]
<img width="1920" height="1030" alt="image" src="https://github.com/user-attachments/assets/70728970-aaf3-403b-8c00-655d92c4d499" />
npm i @svgedit/svgcanvas
then  svgedit.d.ts


```
declare module '@svgedit/svgcanvas' {
    class SvgCanvas {
        constructor(containerElement: string | Element, config: any);
        setMode(mode: string): void;
        $id(canvs): string;
    }
    export default SvgCanvas;
}
```


then  :

```
import SvgCanvas from '@svgedit/svgcanvas';
const { $id, $click, convertUnit } = SvgCanvas as any;
```
